src/LocalBackend/LocalLibrary.vala

```diff
@@ -565,10 +565,20 @@ public class Noise.LocalLibrary : Library {
                 sql.select_add_field ("rowid", null, null);
                 Gda.SqlBuilderId[] ids = null;
 
-                string[] fields = {"title", "artist", "composer", "album_artist", "album", "grouping", "comment"};
+                /* Enabling directory/path/uri search through SearchEntry */
+                string new_search = search;
+                string[] fields = {};
+                if(search.has_prefix("uri:")) {
+                    new_search = search.substring(4);
+                    new_search = new_search.replace(" ", "%20");
```

Comment:
Same here: Is there a reason why [GLib.Uri](https://valadoc.org/glib-2.0/GLib.Uri.html) cannot be used in this case?